### DISCUSSION TITLE
Api gateway host rewrites

### DISCUSSION THREAD

**Discussion by zachhutchins** [mia-platform/community#493] [Q&A]
### What product are you having troubles with?

Console

### What Console version are you using?

1.100.3

### Description

When configuring a proxy in console for external services or in endpoints configuration is there a way to configure it such that we are able to get envoy configuration to have the `auto_host_rewrite` field set, or any other host rewrite options for that matter?

For reference here is the envoy configuration documentation https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-auto-host-rewrite

### Actual Outcome

External service is getting traffic however expects the correct host header to be set in order to internally direct traffic.

### Expected Outcome

Without host rewrite we are unable to route traffic correctly in our external service.

**Comment by swageblock** [comment]
I wanted to post this work around in case others were in need.  We have chosen to use lua to manage the addition of host headers.  This does require the recreation of some of the endpoint definitions, but was the simplest way that we came up with to solve this problem.
The order will be significant as /pre and /prefix would both match for route /prefixes and the first one in the array would be selected.

```
- name: envoy.filters.http.lua.hostrewrite
  listener_name: mylistenerName
  typed_config:
    "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
    inline_code: |
      local host_headers = {
        ["/route1"] = "{{PUBVAR1}}",
        ["/route2"] = "api-{{ENVABBREV}}}.sample.com"
      }
      function envoy_on_request(request_handle)
        local path = request_handle:headers():get(":path")
        if path then
          for prefix, host_header in pairs(host_headers) do
            if string.sub(path, 1, string.len(prefix)) == prefix then
              request_handle:headers():replace(":authority", host_header)
              break
            else
              request_handle:logInfo("No path found in request for listener, path:" .. path)
            end
          end
        else
          request_handle:logInfo("No path extracted from the request")
        end
      end
```
